s3.tf

```diff
@@ -32,3 +36,50 @@ resource "aws_s3_bucket" "this" {
     enabled = var.s3_versioning_enabled
   }
 }
+
+
+data "aws_iam_policy_document" "bucket_policy" {
+  count = var.s3_enabled ? 1 : 0
+
+  statement {
+    sid       = "DenyIncorrectEncryptionHeader"
+    effect    = "Deny"
+    actions   = ["s3:PutObject"]
+    resources = ["arn:aws:s3:::${join("", aws_s3_bucket.this.*.id)}/*"]
+
+    principals {
+      identifiers = ["*"]
+      type        = "*"
+    }
+
+    condition {
+      test     = "StringNotEquals"
+      values   = ["aws:kms"]
+      variable = "s3:x-amz-server-side-encryption"
+    }
+  }
+
+  statement {
+    sid       = "DenyUnEncryptedObjectUploads"
```

Comment:
Will this lock the S3 bucket in case something creates an S3 bucket without KMS encryption with this module?

Reply:
We only create with KMS encryption.